jvm/src/main/scala/cats/jvm/EvalAsync.scala

```diff
@@ -0,0 +1,41 @@
+package cats
+package jvm
+
+import data.Reader
+import java.util.concurrent.{Callable, CountDownLatch, ExecutorService}
+
+object EvalAsync {
+  def apply[A](cb: (A => Unit) => Unit): Eval[A] = Eval.always {
+    val cdl = new CountDownLatch(1)
+    var result: Option[A] = None
+    cb((a: A) => {result = Some(a); cdl.countDown})
+    cdl.await
```

Comment:
TL;DR - sorry for the long message - what I'm saying is that if the design involves any form of blocking threads (like usage of a `CountDownLatch`), for a foundational library like Cats I think the design is wrong.